src/buildercore/core.py

```
@@ -142,22 +152,45 @@ def stack_pem(stackname, die_if_exists=False, die_if_doesnt_exist=False):
 def stack_conn(stackname, username=config.DEPLOY_USER, **kwargs):
     if 'user' in kwargs:
         LOG.warn("found key 'user' in given kwargs - did you mean 'username' ??")
-    data = stack_data(stackname)
+
+    data = stack_data(stackname, ensure_single_instance=True)
```

fails fast if being used over a cluster with multiple EC2 instances